sdks/java/core/src/main/java/org/apache/beam/sdk/io/PubsubIO.java

```diff
@@ -723,84 +666,81 @@ public Duration getMaxReadTime() {
 
       private class PubsubReader extends DoFn<Void, T> {
```

Comment:
Actually, is there a good reason not to switch to `BoundedReadFromUnboundedSource` here?